cpp/tests/transform/bools_to_mask_test.cpp

```diff
@@ -32,7 +32,7 @@ struct MaskToNullTest : public cudf::test::BaseFixture {
   {
     cudf::test::fixed_width_column_wrapper<bool> input_column(
       input.begin(), input.end(), val.begin());
-    std::transform(val.begin(), val.end(), input.begin(), input.begin(), std::logical_and<bool>());
+    std::transform(val.begin(), val.end(), input.begin(), input.begin(), std::logical_and<>());
```

Comment:
What? Um wait, what?

Reply:
This is https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-transparent-functors.html, which is because in C++14 the template arguments for so-called "transparent" functors can be deduced in the right contexts. In this case there isn't really any benefit though and we could go ahead and revert. I believe the primary reason that this functionality was added was to supported heterogeneous lookup when the functors needed to operate on different types. Up to you.

Reply:
[This thread](https://stackoverflow.com/questions/20317413/what-are-transparent-comparators) has some good info.

Reply:
I learned about transparent functors today. I'm good with this change. Thanks.